lib/Service/ReplicateAPIService.php

```diff
@@ -55,6 +58,40 @@ public function createWhisperPrediction(string $audioFileUrl, bool $translate =
 		return $this->request('predictions', $params, 'POST');
 	}
 
+	/**
+	 * Create a prediction and wait for it to complete to return the text result
+	 *
+	 * @param File $file
+	 * @param bool $translate
+	 * @param string $model
+	 * @return string
+	 * @throws LockedException
+	 * @throws NotPermittedException
+	 */
+	public function transcribeFile(File $file, bool $translate = false, string $model = 'large'): string {
+		$prediction = $this->createWhisperPrediction($file->getContent(), $translate, $model);
+		if (isset($prediction['id'])) {
+			$predictionId = $prediction['id'];
+
+			while (isset($prediction['status'])) {
+				if ($prediction['status'] === 'succeeded') {
+					return $translate
+						? $prediction['output']['translation']
+						: $prediction['output']['transcription'];
+				} elseif ($prediction['status'] === 'failed') {
+					throw new Exception('Error transcribing file "' . $file->getName() . '": remote job failed');
+				} elseif ($prediction['status'] === 'canceled') {
+					throw new Exception('Error transcribing file "' . $file->getName() . '": remote job was canceled');
+				} elseif ($prediction['status'] !== 'starting' && $prediction['status'] !== 'processing') {
+					throw new Exception('Error transcribing file "' . $file->getName() . '": unknown prediction status');
+				}
+				sleep(2);
```

Comment:
heh, love it!

Reply:
The `unknown prediction status` message?

Reply:
No, the sleep. We don't get to use sleep often without regretting it :D